public/resources/core/js/services/product-offering.service.js

```diff
@@ -10,52 +10,151 @@
         };
 
-        var PRICE_CURRENCIES = {
-            EUR: 'Euro',
-            USD: 'US Dollar',
-            CAD: 'Canadian Dollar'
+        var TYPES = {
+            CHARGE_PERIOD: {
+                MONTHLY: 'monthly',
+                WEEKLY: 'weekly',
+                YEARLY: 'yearly'
+            },
+            CURRENCY_CODE: {
+                CAD: 'canadian dollar',
+                EUR: 'euro',
+                USD: 'us dollar'
+            },
+            PRICE: {
+                ONE_TIME: 'one time',
+                RECURRING: 'recurring',
+                USAGE: 'usage'
+            }
         };
 
-        var PRICE_PERIODS = {
-            WEEKLY: 'Weekly',
-            MONTHLY: 'Monthly',
-            YEARLY: 'Yearly'
+        var TEMPLATES = {
+            PRICE: {
+                currencyCode: 'EUR',
+                dutyFreeAmount: 0,
+                percentage: 0,
+                taxIncludedAmount: 0,
+                taxRate: 20
+            },
+            PRICEPLAN: {
+                description: '',
+                name: '',
+                price: {},
+                priceType: TYPES.PRICE.ONE_TIME,
+                recurringChargePeriod: '',
+                unitOfMeasure: ''
+            },
+            RESOURCE: {
+                bundledProductOffering: [],
+                category: [],
+                description: '',
+                isBundle: false,
+                lifecycleStatus: LIFECYCLE_STATUS.ACTIVE,
+                name: '',
+                place: [],
+                productOfferingPrice: [],
+                validFor: {},
+                version: '0.1'
+            }
         };
 
-        resource.prototype.getCategories = getCategories;
-        resource.prototype.getPicture = getPicture;
-        resource.prototype.getCheapestPriceplan = getCheapestPriceplan;
-        resource.prototype.formatCheapestPriceplan = formatCheapestPriceplan;
-        resource.prototype.serialize = serialize;
+        var Price = function Price(data) {
+            angular.extend(this, TEMPLATES.PRICE, data);
+            parseNumber(this, ['dutyFreeAmount', 'percentage', 'taxIncludedAmount', 'taxRate']);
+        };
+        Price.prototype.setCurrencyCode = function setCurrencyCode(codeName) {
+
+            if (codeName in TYPES.CURRENCY_CODE) {
+                this.currencyCode = codeName;
+            }
+
+            return this;
+        };
+        Price.prototype.toJSON = function toJSON() {
+            return {
+                currencyCode: this.currencyCode,
+                dutyFreeAmount: this.taxIncludedAmount / ((100 + this.taxRate) / 100),
+                percentage: this.percentage,
+                taxIncludedAmount: this.taxIncludedAmount,
+                taxRate: this.taxRate
+            };
+        };
+        Price.prototype.toString = function toString() {
+            return '' + this.taxIncludedAmount + ' (' + this.currencyCode.toUpperCase() + ')';
```

Comment:
Initial '' is only required when you are only attaching numbers to the string. In this case this is not required since you are adding other string to the string ` (`, `)`...